### ISSUE TITLE
Automatic placement of markers for inline code generation?

### ISSUE THREAD

**Issue by Obbut** [krzysztofzablocki/Sourcery#195]
I'm working on a [object persistence framework](https://github.com/openkitten/meow) for Swift, for which I use Sourcery for code generation. That works really well.

The models that I let users create look like this:

```swift
class MyModel: Model {
    var sample: String
    init() { self.sample = "example" }
    // sourcery:inline:MyModel.Meow
    // sourcery:end
}
``` 

While this is of course great, it would be even greater if I could remove the need for my users to declare the `sourcery:inline` markers by adding them automatically. In the template, there would be a way to define that the inline code generation markers are added (for example) at the end of the class declaration, if they are not existent already. It could look like this: `// sourcery:inline:MyModel`, where Sourcery would look up the type "MyModel" and place the code (and markers) at the bottom of the declaration.

My questions boil down to this:

- Would you be open to a feature like this? I'm willing to implement it myself, but If you don't like it I'd rather save the time
- Would this technically even be (reasonably) possible?
- If yes and yes, can you point me in the right direction for implementing this so I can make a PR?

**Comment by Joannis** [comment]
This would reduce boilerplate code even more than sourcery already does.

**Comment by krzysztofzablocki** [comment]
First off, love your framework idea and its great to see Sourcery leveraged so much 👍 

It's possible for sure, when you parse entries you can get information on when they end so you could track the last `}` of the original type definition.

In your template you could use a new markup to imply the automatic insertion, something like:
`// sourcery:inline:auto:MyModel`

You can take a look at [annotation parser](https://github.com/krzysztofzablocki/Sourcery/blob/master/Sourcery/Parsing/Utils/AnnotationsParser.swift), the replacement logic happens in [here](https://github.com/krzysztofzablocki/Sourcery/blob/master/Sourcery/Sourcery.swift#L301)

Its not that complicated
ps. if you are touching that code it might be worth looking whether you have idea how to fix #156

I'm traveling right now so my availability is limited, but feel free to shoot either me or @ilyapuchka any questions you have 👍

**Comment by Obbut** [comment]
Thanks, I will look into it tomorrow 🙂

**Comment by Obbut** [comment]
I implemented it in #196. I looked into #156 but I'm not able to reproduce it. I haven't had any crashes with inline code generation myself.

Thanks for the pointers to the code, it really helped!
